packages/test-runner-intercept/src/createResolveImport.ts

```diff
@@ -0,0 +1,41 @@
+import { Plugin, ServerStartParams, ResolveOptions, Context } from '@web/dev-server-core';
+
+// Copied from packages/dev-server-core/src/plugins/Plugin.ts as it's not exported
+export type ResolveResult = void | string | { id?: string };
+export interface ResolveImportArguments {
+  source: string;
+  context: Context;
+  code?: string;
+  column?: number;
+  line?: number;
+  resolveOptions?: ResolveOptions;
+}
+export type ResolveImport = (
+  args: ResolveImportArguments,
+) => ResolveResult | Promise<ResolveResult>;
+
+/**
+ * TODO: check if `resolveImport()` can be provied by `@web/dev-server-core`'s API
```

Comment:
just to add options to the table: if it should stay private, we can also make a private shared library in the monorepo that can be used by 2 packages without releasing/exporting to public
it's not the first time I personally run into this, so was curious what others think about that

Reply:
What do you mean with "it should stay private"? Do you mean the comment?

Reply:
I meant `export resolveImport` might not be available via any public export, yet can be reused internally by multiple packages

this is needed sometimes

Reply:
We can avoid this by removing this export altogether. Currently, this export (`resolveImport`) is created to workaround a limitation in the dev server core API. Currently there is no way to continue a resolution in an import. This is something which is widely used in Rollup plugins.
A proposal from Lars was to expose a new API object in the start hook with methods like `resolveImport`. I didn't implement this change (in dev server core) in this PR as it changes some things deep in dev server core as well. This could be added in a next PR though.

Reply:
ok, looks like you are more up-to-date on this than me, feel free to decide here :)